service/matching/matching_engine.go

```diff
@@ -1249,16 +1250,80 @@ func (e *matchingEngineImpl) DescribeTaskQueue(
 			},
 		}, nil
 	}
+
 	// Otherwise, do legacy DescribeTaskQueue
 	partition, err := tqid.PartitionFromProto(req.TaskQueue, request.GetNamespaceId(), req.TaskQueueType)
 	if err != nil {
 		return nil, err
 	}
-	pm, _, err := e.getTaskQueuePartitionManager(ctx, partition, true, loadCauseDescribe)
+	rootPM, _, err := e.getTaskQueuePartitionManager(ctx, partition, true, loadCauseDescribe)
```

Comment:
Did you test this with calling DescribeTaskQueue on child partitions? (i.e. `/_sys/tq/1`)

It "should" be directed to the root but we allow it directed to child partitions so we can get their ack level and stuff. Frontend blocks DescribeTaskQueue on non-root for enhanced mode, but not on default mode. So `rootPM` here is not accurate, this could be a child partition. We have to handle that somehow... maybe we should do LegacyDescribeTaskQueue only in that case and ignore ReportStats? Or frontend should block calls with ReportStats to non-root?

Reply:
👍 renamed it back to `pm`. I added the "invalid argument" error as discussed in the thread for `TestNonRoot`.